standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionInfo.java

```diff
@@ -277,6 +277,6 @@ public static CompactionInfo optionalCompactionInfoStructToInfo(OptionalCompacti
 
   public void setWriteIds(Set<Long> writeIds) {
     this.writeIds = writeIds;
-    isSetWriteIds = true;
+    isSetWriteIds = !writeIds.isEmpty();
```

Comment:
Why do we maintain 2 states for this anyway?
We could have a method instead:
public boolean isSetWriteIds() { return !writeIds.isEmpty(); }